api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java

```diff
@@ -220,6 +227,21 @@ public WorkspaceAuditLogQueryResponse getAuditLogEntries(
         workspaceDatabaseId, limit, after, before);
   }
 
+  @Override
+  public String getReadOnlyNotebook(
```

Comment:
Can you please add a unit test on this service (and possibly notebooks service)? I'm especially interested in showing this serialization workaround.